dashboard/src/components/ChartView/ChartDeployButton.tsx

```diff
@@ -1,42 +1,38 @@
+import { push } from "connected-react-router";
 import * as React from "react";
-import { Redirect } from "react-router";
-import { IChartVersion } from "../../shared/types";
+import { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { IChartVersion, IStoreState } from "../../shared/types";
 import { app } from "../../shared/url";
 
-interface IChartDeployButtonProps {
+export interface IChartDeployButtonProps {
   version: IChartVersion;
   namespace: string;
 }
 
-interface IChartDeployButtonState {
-  clicked: boolean;
-}
-
-class ChartDeployButton extends React.Component<IChartDeployButtonProps, IChartDeployButtonState> {
-  public state: IChartDeployButtonState = {
-    clicked: false,
-  };
+function ChartDeployButton(props: IChartDeployButtonProps) {
+  const [clicked, setClicked] = useState(false);
+  const setClickedTrue = () => setClicked(true);
 
-  public render() {
-    const { version } = this.props;
-    const { namespace } = this.props;
-    const versionStr = version.attributes.version;
+  const currentCluster = useSelector((state: IStoreState) => state.clusters.currentCluster);
+  const dispatch = useDispatch();
 
-    return (
-      <div className="ChartDeployButton text-r">
-        <button className="button button-primary button-accent" onClick={this.handleClick}>
-          Deploy
-        </button>
-        {this.state.clicked && (
-          <Redirect push={true} to={app.apps.new(version, namespace, versionStr)} />
-        )}
-      </div>
+  if (clicked) {
+    const newAppURL = app.apps.new(
+      props.version,
+      currentCluster,
+      props.namespace,
+      props.version.attributes.version,
     );
+    dispatch(push(newAppURL));
```

Comment:
Switched to a dispatch as otherwise we also need to wrap the component being tested in a `<Router>` (to use Redirect - tests failed because the Redirect should not be used outside a `Router`, though not sure why the change to a function component caused that. It seems cleaner too (dispatching / calling a function, though haven't yet tested IRL)

Reply:
that's what we do in the containers so it makes sense:
```
src/containers/AppNewContainer/AppNewContainer.tsx:56:    push: (location: string) => dispatch(push(location)),
```